### DISCUSSION TITLE
Nested material top tabs navigator loses index state when navigating parent navigator

### DISCUSSION THREAD

**Discussion by tibbe** [react-navigation/react-navigation#11554] [Q&A]
I have a material top tabs navigator nested inside one of the screen of a stack navigator. When I navigate away from the top tabs screen the nested navigator resets its state, even though it's still on the stack of the stack navigator. This causes the `goBack()` call on the next screen to go back to the tabs screen with the first tab selected, instead of whatever tab was selected when we navigated. Using the Flipper plugin I verified this by looking at the `index` state kept for the nested navigator.

Before navigation state:

```
index: 1
routes: [SomeScreen, ScreenWithTabs(index: 1)]
```

After navigation state:

```
index: 2
routes: [SomeScreen, ScreenWithTabs(index: 0), SomeOtherScreen]
```

Observe how the `index` of the nested navigator got reset by pushing another screen onto the stack.

How can I avoid the nested navigator state getting reset?

**Comment by tibbe** [comment]
I observed that the problem goes away if I remove these lines from the `ScreenWithTabs`:

```ts
  useEffect(() => {
    const unsubscribe = navigation.addListener('transitionEnd', () => {
      navigation.setParams({
        animation: true,
      });
    });

    return unsubscribe;
  }, [navigation]);
```

It seems that the call to `setParams` isn't shallow-merged as expected but also affects the `index` somehow.

**Comment by tibbe** [comment]
If I replace `setParams` above with

```ts
      navigation.dispatch({
        ...CommonActions.setParams({
          screen: 'ScreenWithTabs',
          params: {animation: true},
        }),
      });
```
things work. I wonder if `setParams` for `transitionEnd` affected the wrong screen somehow or otherwise caused a navigator state reset.

**Comment by Nikualla1** [comment]
I have the same issue, but it's only happening on Android for some reason.

**Reply by itsnyx** [reply]
did you find any solution for this ?

**Comment by sayan-avantai** [comment]
Any solution?
